modules/frontend/v1/frontend.go

```diff
@@ -118,6 +118,7 @@ func New(cfg Config, limits Limits, log log.Logger, registerer prometheus.Regist
 }
 
 func (f *Frontend) starting(ctx context.Context) error {
+	f.subservicesWatcher = services.NewFailureWatcher()
```

Comment:
i'm not finding where this is initialized in our current codebase. how is this not panicing with a nil pointer?

Reply:
You can still access struct methods on nil pointers. This test will run just fine

```go
func TestAccessMethodOnNilPointer(t *testing.T) {
	m, err := services.NewManager(services.NewIdleService(nil, nil))
	require.NoError(t, err)

	var fw *services.FailureWatcher
	assert.Nil(t, fw)
	fw.WatchManager(m)
}
```